clippy_lints/src/doc/needless_doctest_main.rs

```diff
@@ -120,6 +127,17 @@ pub fn check(
 
     let trailing_whitespace = text.len() - text.trim_end().len();
 
+    // We currently only test for "fn main". Checking for the real
+    // entrypoint (with tcx.entry_fn(())) in each block would be unnecessarily
+    // expensive, as those are probably intended and relevant. Same goes for
+    // macros and other weird ways of declaring a main function.
+    //
+    // Also, as we only check for attribute names and don't do macro expansion,
+    // we can check only for #[test]
+    if !(text.contains("fn main") || text.contains("#[test]")) {
```

Comment:
hm, if someone writes `fn  main()` instead of `fn main()` will this cause false negatives?

Reply:
Fixed (and improved filtering!) Now we check for instances of `fn` (two times) and `main` in general.